consensus/hotstuff/voteaggregator/vote_aggregator_test.go

```diff
@@ -575,6 +575,45 @@ func (as *AggregatorSuite) TestDuplicateVotesBeforeBlock() {
 	require.NoError(as.T(), err)
 }
 
+// UNHAPPY PATH
+// TestEquivocation_DoubleProposal tests that VoteAggregator handles a double
+// proposal equivocation correctly. There are two ways, we can feed a double
+// proposal into the VoteAggregator:
+//   *  VoteAggregator.BuildQCOnReceivedBlock
+//   *  VoteAggregator.StoreVoteAndBuildQC
+// We test that both handle the double proposal gracefully (without error).
+func (as *AggregatorSuite) TestEquivocation_DoubleProposal() {
+	testView := uint64(5)
+	bp1 := newMockBlock(as, testView, as.participants[0].NodeID)
+	bp2 := newMockBlock(as, testView, as.participants[0].NodeID)
+
+	// Each of the blocks contain the proposer's signature. Hence, the proposer
+	// votes for both of its conflicting blocks => expect double vote notification
+	as.notifier.On("OnDoubleVotingDetected", bp1.ProposerVote(), bp2.ProposerVote()).Return()
+
+	as.aggregator.StoreProposerVote(bp1.ProposerVote())
+	qc, built, err := as.aggregator.BuildQCOnReceivedBlock(bp1.Block)
+	require.Nil(as.T(), qc)
+	require.False(as.T(), built)
```

Comment:
Is this false just because there is only one vote for the block? Or is it related to the fix as well?

Reply:
In the test setup, we have 7 consensus participants, i.e. we require 5 votes for a supermajority. The fix didn't change this logic.